src/lib.rs

```diff
@@ -1017,14 +1028,14 @@ pub mod pallet {
 		/// - `nonce`: The current value count for the query identifier.
 		/// - `query_data`: The data used to fulfil the data query.
 		#[pallet::call_index(7)]
-		#[pallet::weight(386325000)]
+		#[pallet::weight(<T as Config>::WeightInfo::submit_value(T::MaxQueryDataLength::get(), T::MaxValueLength::get()))]
```

Comment:
Minor thing, but can the weight function parameters have the same order as extrinsic please? E.g. value, query_data